spec/consumer_resource_spec.rb

```diff
@@ -1553,6 +1551,82 @@ def verify_installed_pids(consumer, product_ids)
     }.to raise_exception(RestClient::BadRequest)
   end
 
+  it 'should create consumer with multiple environments' do
+    user_cp = user_client(@owner1, random_string('billy'))
+    env1 = @cp.create_environment(@owner1['key'], random_string("env_id-1"), random_string("env_name-1"))
+    env2 = @cp.create_environment(@owner1['key'], random_string("env_id-2"), random_string("env_name-2"))
+
+    consumer = user_cp.register(random_string('system'), :system, nil, {}, nil, nil, [], [], [{'id' => env1.id}, {'id' => env2.id}])
+    consumer_client = Candlepin.new(nil, nil, consumer['idCert']['cert'], consumer['idCert']['key'])
+    expect(consumer.environments.size).to eq(2)
```

Comment:
The priority of environments should be preserved not only when updating a consumer, but also when registering. Let's add this assertion to the end of this test

Reply:
Done.